Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -2898,6 +2902,107 @@ bool Element::has_paint_containment() const
     return false;
 }
 
+int Element::number_of_owned_list_items()
+{
+    auto number_of_owned_li_elements = 0;
+    for_each_child_of_type<DOM::Element>([&](auto& child) {
+        if (child.list_owner() == this) {
+            number_of_owned_li_elements++;
+        }
+        return IterationDecision::Continue;
+    });
+    return number_of_owned_li_elements;
+}
+
+// https://html.spec.whatwg.org/#list-owner
+Element* Element::list_owner()
+{
+    // Any element whose computed value of 'display' is 'list-item' has a list owner, which is determined as follows:
+    if (!computed_properties()->display().is_list_item())
+        return nullptr;
+
+    // FIXME: 1. If the element is not being rendered, return null; the element has no list owner.
```

Comment:
Elsewhere, we check whether an element has a layout node, to check whether it is being rendered. You will need to call `document->update_layout()` before doing any layout node checks to ensure the layout tree is up to date.

Reply:
I tried that first, but it doesn't work because this function is called during a layout update. I have revised the PR so that the text of the ListItemMarkerBox is calculated lazily. Now a layout update should no longer be necessary.